cmd/gardenlet/app/gardenlet.go

```diff
@@ -278,20 +280,32 @@ func NewGardenlet(ctx context.Context, cfg *config.GardenletConfiguration) (*Gar
 		return nil, err
 	}
 
-	k8sGardenClient, err := kubernetes.NewWithConfig(
-		kubernetes.WithRESTConfig(restCfg),
-		kubernetes.WithClientOptions(client.Options{
-			Scheme: kubernetes.GardenScheme,
-		}),
-	)
+	gardenClientMapBuilder := clientmapbuilder.NewGardenClientMapBuilder().
+		WithRESTConfig(restCfg)
+	seedClientMapBuilder := clientmapbuilder.NewSeedClientMapBuilder().
+		WithInCluster(cfg.SeedSelector == nil).
+		WithClientConnectionConfig(&cfg.SeedClientConnection.ClientConnectionConfiguration)
+	shootClientMapBuilder := clientmapbuilder.NewShootClientMapBuilder().
+		WithClientConnectionConfig(&cfg.ShootClientConnection.ClientConnectionConfiguration)
+
+	clientMap, err := clientmapbuilder.NewDelegatingClientMapBuilder().
+		WithGardenClientMapBuilder(gardenClientMapBuilder).
+		WithSeedClientMapBuilder(seedClientMapBuilder).
+		WithShootClientMapBuilder(shootClientMapBuilder).
```

Comment:
Also looks like `WithLogger(logger).` needs to be specified for this builder as otherwise the gardenlet start fails with `Error: failed to build ClientMap: logger is required but not set`.

Reply:
Good catch!

Reply:
Done, gcm and gardenlet should be starting up just fine again.